scripts/extensions.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+# Install and/or enable extensions.
+# Usage:
+#
+#   extensions.sh PHP_VERSION LIST_OF_EXTENSIONS
+
+set -e
+
+STATIC_EXTENSIONS=(sqlsrv swoole)
+
+function install_packaged_extensions {
+    local EXTENSIONS=${@:1}
+    echo "Installing packaged extensions: ${EXTENSIONS}"
+    apt update
+    apt install -y "${EXTENSIONS}"
+}
+
+function enable_static_extension {
+    local PHP=$1
+    local EXTENSION=$2
+    echo "Enabling ${EXTENSION} extension"
+    phpenmod -v "${PHP}" -s ALL "${EXTENSION}"
+}
+
+function enable_sqlsrv {
+    local __result=$1
+    local PHP=$2
+    local EXTENSIONS=${@:3}
+    if [[ ! ${PHP} =~ (7.3|7.4|8.0) ]];then
+        echo "Skipping enabling of swoole extension; not supported on PHP < 7.3"
+        eval $__result="${EXTENSIONS}"
+    else
+        enable_static_extension "${PHP}" sqlsrv
+		eval $__result=$(echo "${EXTENSIONS}" | sed -E -e 's/php[0-9.]+-(pdo[_-]){0,1}sqlsrv/ /g' | sed -E -e 's/\s{2,}/ /g')
+    fi
+}
+
+function enable_swoole {
+    local __result=$1
+    local PHP=$2
+    local EXTENSIONS=${@:3}
+    if [[ ! ${PHP} =~ (7.3|7.4|8.0) ]];then
```

Comment:
I wonder if it would be possible to detect php versions automatically. adding 8.1 should not require us to change multiple files.

Reply:
We need to do this manually. Extensions may or may not be updated to a particular ABI, so if we automatically enable for a new version, we may actually have errors when building the container due to extension compilation failure.

Consider this: MS only provides the sqlsrv binaries on Ubuntu 20.04 for 7.3, 7.4, and 8.0; we don't know if/when they'll have them available for 8.1. What if we build the day that 8.1 releases, and then this attempts to enable the extension for 8.1?

Similarly, many third-party extensions like Swoole take a few days to a few months to update. Automatically updating that list can lead to errors.